mifosng-provider/src/main/java/org/mifosplatform/portfolio/charge/service/ChargeWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -82,8 +86,17 @@ public CommandProcessingResult createCharge(final JsonCommand command) {
             if (glAccountId != null) {
                 glAccount = this.gLAccountRepository.findOneWithNotFoundDetection(glAccountId);
             }
+           
+            final Charge charge = Charge.fromJson(command, glAccount) ;
+           
+          //MIFOSX-Modification in the charge definition for TrancheDisbursement.
+            if (ChargeTimeType.TRANCHE_DISBURSEMENT.getValue().equals(charge.getChargeTimeType())) {
```

Comment:
Move this code to a private method and perform this validation. Throw appropriate exception